server/handler/base.go

```diff
@@ -391,6 +401,31 @@ func selectionToReviewersRequest(s reviewer.Selection) github.ReviewersRequest {
 	return req
 }
 
+func (b *Base) dismissStaleReviewsForResult(ctx context.Context, prctx pull.Context, result common.Result) error {
+	if len(result.DiscardedReviews) == 0 {
+		return nil
+	}
+
+	for _, d := range result.DiscardedReviews {
+		if d.State != pull.ReviewApproved {
```

Comment:
I think passing the review state through to here is unnecessary. When computing the list of candidates, reviews are pre-filtered to only have the correct state (approval in this case.) A non-approved review would not be a candidate for granting approval.

Reply:
ahh right, thanks.